main.go

```diff
@@ -190,14 +193,24 @@ func generate(in configInput) ([]byte, error) {
 		},
 	}
 
+	if in.configScope != "" {
+		c.Node.Metadata["TRAFFICDIRECTOR_SCOPE_NAME"] = in.configScope
+	}
+
 	for k, v := range in.metadataLabels {
 		c.Node.Metadata[k] = v
 	}
 	if in.includeV3Features {
 		// xDS v2 implementation in TD expects the projectNumber and networkName in
 		// the metadata field while the v3 implementation expects these in the id
 		// field.
-		c.Node.Id = fmt.Sprintf("projects/%d/networks/%s/nodes/%s", in.gcpProjectNumber, in.vpcNetworkName, uuid.New().String())
+		var networkIdentifier string
+		if in.configScope != "" {
+			networkIdentifier = fmt.Sprintf("scope:%s", in.configScope)
```

Comment:
Should we restrict the config scope to only certain characters?

Reply:
Good point. Doing a little digging, it looks like scope has a maximum length of 64 and must conform to the regex `[a-zA-Z][-a-zA-Z0-9]*`. This validation is done in CLH, but it wouldn't hurt to add it here as well. Done.